File: TestDoubles/java/PoS_before/src/main/java/POSTerminal.java
public class POSTerminal {
    CCVerifier ccVerifier;

    POSTerminal() {
        this.ccVerifier = CCVerifier.getInstance();
    }

    void verifySale(double amount) throws InvalidChargeException {
        if (!ccVerifier.approveCharge(amount)) {
            throw new InvalidChargeException();
        }
    }

    public class InvalidChargeException extends Throwable {
        public InvalidChargeException() {
            super("Charge exceeds single transaction amount or credit limit");
        }
    }
}

File: TestDoubles/csharp/PoS_before/POSTerminal/PosTerminal.cs
using System;

namespace PosExample
{
    public class PosTerminal
    {
        readonly CcVerifier _ccVerifier;

        public PosTerminal()
        {
            this._ccVerifier = CcVerifier.GetInstance();
        }
        
        void VerifySale(double amount)
        {
            if (!_ccVerifier.ApproveCharge(amount))
            {
                throw new InvalidChargeException();
            }
        }
    }

    internal class InvalidChargeException : Exception
    {
        public InvalidChargeException() : 
            base("Charge exceeds single transaction amount or credit limit")
        {
        }
    }
}